lwgenerate/gcp/gcp.go

```diff
@@ -0,0 +1,521 @@
+package gcp
+
+import (
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/pkg/errors"
+
+	"github.com/lacework/go-sdk/lwgenerate"
+)
+
+type ExistingServiceAccountDetails struct {
+	// Existing Service Account Name
+	Name string
+
+	// Existing Service Account private key in JSON format, base64 encoded
+	PrivateKey string
+}
+
+// NewExistingServiceAccountDetails Create new existing Service Account details
+func NewExistingServiceAccountDetails(name string, privateKey string) *ExistingServiceAccountDetails {
+	return &ExistingServiceAccountDetails{
+		Name:       name,
+		PrivateKey: privateKey,
+	}
+}
+
+type GenerateGcpTfConfigurationArgs struct {
+	// Should we configure AuditLog integration in LW?
+	AuditLog bool
+
+	// Should we configure CSPM integration in LW?
+	Config bool
+
+	// Path to service account credentials to be used by Terraform
+	ServiceAccountCredentials string
+
+	// Should we configure an Organization wide integration?
+	OrganizationIntegration bool
+
+	// Supply a GCP Organization ID, only asked if OrganizationIntegration is True
+	GcpOrganizationId string
+
+	// Supply a GCP Project ID, to host the new resources
+	GcpProjectId string
```

Comment:
Would it be fair to make this a required field in the context of running this via the CLI, rather than relying on the end-user to pass this using env vars?
The Project ID must be set one way or another, but for the purposes of building the TF via the CLI it would be nice to require this to be passed in always, and then we can set it in the google provider.
I'd appreciate any input on this @afiune @ipcrm @dmurray-lacework @mikelaramie

Reply:
This would result in us building the googler Terraform provider like so
```
provider "google" {
  credentials = "/path/to/credentials"
  project     = "project1"
}
```

Reply:
W.r.t. `project` in the provider: I love this idea unless we're considering creating the project as part of this workflow.  Makes it a lot easier than specifying `project=` in each resource.  

W.r.t `credentials`: Are we requiring a service account, or is that credentials field optional?  Ideally we should default to gcloud/application default credentials for auth but provide the option for a SA if necessary.